static_frame/core/frame.py

```diff
@@ -1493,42 +1495,66 @@ def from_sql(cls,
             {name}
             {consolidate_blocks}
         '''
-        row_gen = connection.execute(query)
+        sql_iterable = connection.execute(query)
 
-        own_columns = False
         columns = None
-        if columns_depth == 1:
-            columns = cls._COLUMNS_CONSTRUCTOR(b[0] for b in row_gen.description[index_depth:])
-            own_columns = True
-        elif columns_depth > 1:
-            # use IH: get via static attr of columns const
-            constructor = cls._COLUMNS_HIERARCHY_CONSTRUCTOR.from_labels_delimited
-            labels = (b[0] for b in row_gen.description[index_depth:])
-            columns = constructor(labels, delimiter=' ')
+        own_columns = False
+        filter_row = lambda row: row
+
+        if columns_depth >= 1:
+            labels = (col for (col, *_) in sql_iterable.description[index_depth:])
             own_columns = True
 
+            if columns_depth == 1:
+                if columns_select:
+                    indices, labels = map(tuple, zip(*((i, label) for (i, label) in enumerate(labels) if label in columns_select)))
```

Comment:
As we are using `in` on a list, maybe better to maybe convert `columns_select` to a set before this move?